api/python/t4/packages.py

```diff
@@ -143,6 +135,46 @@ def _verify_hash(self, read_bytes):
         if digest != self.hash.get('value'):
             raise QuiltException("Hash validation failed")
 
+    def _set_path(self, path, meta=None):
+        """
+        Sets the path for this PackageEntry.
+        """
+        with open(path, 'rb') as file_to_hash:
+            hash_obj = {
+                'type': 'SHA256',
+                'value': hash_file(file_to_hash)
+            }
+
+        self.size = os.path.getsize(path)
+        self.physical_keys = [pathlib.Path(path).resolve().as_uri()]
+        self.hash = hash_obj
+        if meta is not None:
+            self.set_user_meta(meta)
+        return self
+
+    def set(self, path=None, meta=None):
+        """
+        Returns self with the physical key set to path.
+
+        Args:
+            logical_key(string): logical key to update
+            path(string): new path to place at logical_key in the package
+                Currently only supports a path on local disk
+            meta(dict): metadata dict to attach to entry. If meta is provided, set just
+                updates the meta attached to logical_key without changing anything
+                else in the entry
+
+        Returns:
+            self
+        """
+        if path is None and meta is None:
+            raise PackageException('Must specify either path or meta')
+
+        if path is None:
+            return self.set_user_meta(meta)
```

Comment:
Looks weird cause `set_user_meta` doesn't return anything.

I'd do something like this, to make it more uniform:
```
if path is not None:
  ...
if meta is not None:
  ...
```

Also, now that I think about, it's kinda weird to have:
- `set`
- `set_user_meta`
- `_set_path`
Maybe it should be `set_path`. Or maybe we should get rid of `set_user_meta` cause `set` is sufficient.

Reply:
Restructured it to follow your advice, but leaving method signatures for now. I think `set` and `set_user_meta` were minimum API requirements.